migrate to iris dataset

dquerales opened the pull request:
This is a complete overhaul for iris dataset

dquerales committed:
migrate to iris dataset

vargas88hugo commented on api.py:
```diff
@@ -18,4 +18,4 @@ class ScoringItem(BaseModel):
     yhat = model.predict(df)
-    return {"prediction":int(yhat)}
\ No newline at end of file
+    return {"prediction":str(yhat)}
```
add new line

dquerales replied:
can you give an example of what are you meaning?

vargas88hugo replied:
<img width="41" alt="image" src="https://github.com/dquerales/FastAPI/assets/51675103/8fa458ba-de1d-4b5b-9209-fb888bad15b2">

this red circle with a hyphen inside of it indicates that the file doesn't end with an EOL or end of line. Different computer systems use different line ending styles. Some use a simple newline character (\n) such as Linux, while others use this one (\r\n) like Windows. Adding that EOL makes sure your code plays nice on all systems with all kind of tools. It's also a way to ensure consistency and promote good practices among all files

vargas88hugo replied:
you just need to add an empty new line after the last line of code

vargas88hugo replied:
it would be a good idea to implement a linter into a github action to validate that all files are consistent across the project. This way, we can define a unique style to improve consistency and readability

dquerales replied:
Sure. I am about to add a new github action for this

vargas88hugo commented on api.py:
```diff
@@ -6,10 +6,10 @@
+    petal_length: float 
+    petal_width: float
 
 with open('model/model.pkl', 'rb') as f: 
     model = pickle.load(f)
```
Add exception to handle possible errors
```suggestion
try:
    with open('model/model.pkl', 'rb') as f: 
        model = pickle.load(f)
except Exception as e:
    logging.error(f"Error loading model: {e}")
    raise HTTPException(status_code=500, detail="Model loading error")
```

dquerales replied:
Sure, this changes are already tested?

vargas88hugo replied:
This was an example of commenting and suggesting changes in several lines, but it was not tested. We would need to standardize error handling across the project. A document would be a good fit.

dquerales replied:
This code seems to dont work. It will be adressed in the next pr

vargas88hugo commented on src/train_model.py:
```diff
@@ -1,17 +1,23 @@
+print('Logistic Regression \n')
+print('accuracy: ', accuracy_score(y_test, y_pred))
+
+pickle.dump(model, open('../model/model.pkl', 'wb'))
```
same here

dquerales committed:
feat: add pylint github actions

dquerales committed:
fix: add github action python to install requirements

dquerales committed:
test: yaml sintax again

dquerales committed:
test: github actions fix

dquerales commented:
New pylint actions was added to repository.

dquerales commented on .github/workflows/pylint.yml:
This is the new yml for pylint github action

dquerales commented on .github/workflows/pylint.yml:
```diff
@@ -0,0 +1,24 @@
+        pip install -r requirements.txt
+    - name: Analysing the code with pylint
+      run: |
+        pylint $(git ls-files '*.py')
```
\n

dquerales replied:
resolved

dquerales committed:
style: formatting

dquerales committed:
style: reformatting

dquerales committed:
test: add linters

dquerales committed:
style: add black yml

dquerales committed:
style: black formatting

dquerales committed:
actions: test

dquerales committed:
test: test action again

dquerales committed:
test: actions

dquerales committed:
test: test action

dquerales committed:
actions: fix

dquerales committed:
actions: test new actions

dquerales committed:
prevent new branches

dquerales committed:
actions: new commit changes

dquerales committed:
[AUTO]:art: Format Python code with psf/black

Files changed:
M	api.py
M	src/config.py
M	src/train_model.py
M	test/test_api.py

dquerales committed:
actions: remove pylint for black

dquerales committed:
[AUTO]:art: Format Python code with psf/black

Files changed:
M	api.py
M	test/test_api.py

dquerales committed:
actions: fix

dquerales committed:
Merge branch 'test' of github.com:dquerales/FastAPI into test

dquerales reviewed (commented):
Review finished. I cant approve my own pull request @vargas88hugo

vargas88hugo commented on .github/workflows/black.yml:
```diff
@@ -0,0 +1,23 @@
+          # add_timestamp: true
+          commit_prefix: "[AUTO]"
+          commit_message: ":art: Format Python code with psf/black"
+          # force: false
+          # target_branch: update/version # to create a new branch
```
unnecessary comments?

dquerales replied:
Fixed

vargas88hugo commented on api.py:
```diff
@@ -1,21 +1,24 @@
-async def scoring_endpoint(item:ScoringItem): 
+
+@app.post("/")
+async def scoring_endpoint(item: ScoringItem):
```
should we configure some validation and error handling to set the correct status code for specific cases? At this moment, we can use the following ones:

* 500 Internal Server Error: A generic error message indicating that something went wrong on the server's end, but the server couldn't provide specific details.
* 400 Bad Request: The server could not understand the request due to invalid syntax or missing parameters.

Eventually we will have more cases

vargas88hugo replied:
the method name in a controller should be very similar than the API definition
```suggestion
async def scoring_endpoint(item: ScoringItem):
```

vargas88hugo commented on src/train_model.py:
```diff
@@ -1,17 +1,25 @@
+y = df[config.TARGET]
 
-model = DecisionTreeClassifier()
+X_train, X_test, y_train, y_test = train_test_split(
```
should we have a naming convention? for example, X_train starts with an uppercase, but y_train with a lowercase. In my opinion, variables should always begin with a lowercase, unless  they are constants or classes

vargas88hugo commented on api.py:
```diff
@@ -1,21 +1,24 @@
-@app.post('/')
-async def scoring_endpoint(item:ScoringItem): 
+
+@app.post("/")
```
what if we define our API? Something like `domain/id/action`. This case could be something like `model/predict`, it could be better so I'm open to suggestions

vargas88hugo reviewed (changes requested):
Great job! Some minor comments, but I love the github action, I want to see it in action

vargas88hugo commented on api.py:
```diff
@@ -1,21 +1,24 @@
+
+@app.post("/")
+async def scoring_endpoint(item: ScoringItem):
     df = pd.DataFrame([item.dict().values()], columns=item.dict().keys())
```
it's possible that this goes against the Python philosophy, but what about more descriptive names? The code should be self-documented, would data_frame be better? Not everyone has the same context as you
```suggestion
    data_frame = pd.DataFrame([item.dict().values()], columns=item.dict().keys())
```

vargas88hugo commented on api.py:
```diff
@@ -1,21 +1,24 @@
+@app.post("/")
+async def scoring_endpoint(item: ScoringItem):
     df = pd.DataFrame([item.dict().values()], columns=item.dict().keys())
     yhat = model.predict(df)
```
yhat what means?

vargas88hugo commented on src/train_model.py:
```diff
@@ -1,17 +1,25 @@
+df = pd.read_csv(config.SOURCE)
 
-X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
+X = df.drop([config.TARGET], axis=1)
```
what is X?

vargas88hugo commented on src/train_model.py:
```diff
@@ -1,17 +1,25 @@
 
-X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
+X = df.drop([config.TARGET], axis=1)
+y = df[config.TARGET]
```
what is y?

vargas88hugo commented on src/train_model.py:
```diff
@@ -1,17 +1,25 @@
+model = LogisticRegression()
 model.fit(X_train, y_train)
 
 y_pred = model.predict(X_test)
```
what is y_pred, prediction, predator or what else?

vargas88hugo commented on src/train_model.py:
```diff
@@ -1,17 +1,25 @@
-accuracy = accuracy_score(y_test, y_pred)
-print(f"Model Accuracy: {accuracy}")
+print("Logistic Regression \n")
+print("accuracy: ", accuracy_score(y_test, y_pred))
```
this is only a suggestion, I know that I have been very nitpicky, but we should implement level logs. Even we could create a class for that. Here some examples of level logs:

DEBUG: used to debug
INFO: useful to provide information about an operation
WARNING: it's not an error, but it warns potential issues
ERROR: errors that needs attention
CRITICAL: critical error

dquerales committed:
actions: add train model as github actions

dquerales committed:
[AUTO]:art: Format Python code with psf/black

Files changed:
M	api.py
M	test/test_api.py

dquerales committed:
test: test multiple actions

dquerales committed:
actions: secuential actions

dquerales committed:
fix

dquerales committed:
more fixes

dquerales committed:
run after train

dquerales committed:
more test

dquerales committed:
archive action

dquerales committed:
test

dquerales committed:
more changes

dquerales committed:
fix

dquerales committed:
more fixes

dquerales committed:
more more fixes

actions-user committed:
add model.pkl

dquerales commented:
The current changes are merged to main branch. The other changes will be adressed in the next wave of development
